src/value.c

```diff
@@ -755,8 +779,19 @@ int RSValue_SendReply(RedisModule_Reply *reply, const RSValue *v, SendReplyFlags
     case RSValue_Null:
       return RedisModule_Reply_Null(reply);
 
-    case RSValue_Duo: {
-      return RSValue_SendReply(reply, RS_DUOVAL_OTHERVAL(*v), flags);
+    case RSValue_JSON: {
+      if (!RS_JSONVAL_SERIALIZED(*v)) {
+        RedisModuleString *serialized;
+        japi->getJSONFromIter(RS_JSONVAL_ITER(*v), reply->ctx, &serialized);
```

Comment:
Is this thread safe? we might not have the GIL held at this point. What happens to an iterator if the JSON key is modified (update/delete)?

Reply:
The iterator is independent from the JSON key, so we're fine in that sense (results are computed in iterator construction time).
But, multiple threads using the same iterator concurrently can be a problem.
Checking it out, Tnx.

Reply:
This isn't. Good catch Tnx